sql/set_var.h

```diff
@@ -308,6 +309,10 @@ class sys_var {
 
   void save_default(THD *thd, set_var *var) { global_save_default(thd, var); }
 
+  virtual void persist_only_to_string(THD *thd, set_var *var, String *dest) {
+    saved_value_to_string(thd, var, dest->c_ptr());
```

Comment:
Don't we have the string buffer issue? Why not pass dest to save_value_to_string()